matlab_simulation/08-planning/example_search_graph.m

```diff
@@ -0,0 +1,47 @@
+% Get same problem every time (fix random number generator start point)
+rng('default')
+
+%% Generate a graph
+n = 30; % Nodes
+s = 100; % Space
+% Random nodes
+nodes = s*rand(n,2);
+e = sparse(zeros(n,n));
+D = sparse(zeros*ones(n,n));
+
+% Add closest p edges
+p = 5;
+for i = 1:n
+    for j = 1:n
+        d(j) = norm(nodes(i,:)-nodes(j,:));

```

Comment:
Would the function pdist2 be more useful when calculating distances between sets of points instead of the for loop ? http://www.mathworks.com/help/stats/pdist2.html. That way we dont have to use for loops